### PR TITLE
chore(deps): bump lua-resty-openssl and lua-kong-nginx-module to 1.3.1 and 0.11.0

### PR DESCRIPTION
**Pull request by Water-Melon**
### Summary

- Add a C function and a lua function for `get_socket_ssl` in `lua-kong-nginx-module`.
- Use the C functions provided by lua-kong-nginx-module as an alternative to the auxiliary C functions in lua-resty-openssl.

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-3791

### CODE REVIEW

**Review by ADD-SP — COMMENTED** [review]
We'd better to create two separated commits for each dependency.

**Comment by Water-Melon** [comment]
> We'd better to create two separated commits for each dependency.

Alright, this PR is to bump `lua-kong-nginx-module`.

**Comment by ADD-SP** [comment]
> > We'd better to create two separated commits for each dependency.
> 
> Alright, this PR is to bump `lua-kong-nginx-module`.

Actually, you can create two commits in a single PR, and I can click `Rebase and merge`.

**Review by bungle — APPROVED** [review]

**Comment by fffonion** [comment]
let's paste the changelog between the old version and new version for each dependency, and put in the PR description.

Ideally, the changelog would also appear in the commit message, if it's not too long.

**Review by oowl — APPROVED** [review]

**Comment by bungle** [comment]
Please merge this first:
https://github.com/Kong/kong-ee/pull/8901

It includes Kongrocks bump that also includes latest resty-openssl.

File: changelog/unreleased/kong/bump-lua-kong-nginx-module.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
-message: "Bumped lua-kong-nginx-module from 0.8.0 to 0.10.0"
+message: |
+  Bumped lua-kong-nginx-module from 0.8.0 to 0.10.0
+  Bumped lua-kong-nginx-module from 0.10.0 to 0.11.0
```
Comment (ADD-SP) [inline changelog/unreleased/kong/bump-lua-kong-nginx-module.yml:3]:
``` suggestion
message: |
  Bumped lua-kong-nginx-module from 0.8.0 to 0.11.0
```

File: changelog/unreleased/kong/bump-lua-resty-openssl.yml [hunk 2]
```diff
@@ -1,3 +1,5 @@
-message: "Bumped lua-resty-openssl to 1.2.1"
+message: |
+  Bumped lua-resty-openssl to 1.2.1
+  Bumped lua-resty-openssl to 1.3.1
```
Comment (ADD-SP) [inline changelog/unreleased/kong/bump-lua-resty-openssl.yml:3]:
``` suggestion
message: |
  Bumped lua-resty-openssl from 1.2.0 to 1.3.1
```

File: changelog/unreleased/kong/bump-lua-kong-nginx-module.yml (hunk 1 above)
Comment (Water-Melon) [reply changelog/unreleased/kong/bump-lua-kong-nginx-module.yml]: updated

**Review by fffonion — APPROVED** [review]

**Review by ADD-SP — APPROVED** [review]

**Pull request merged**